pkg/operator/reports.go

```diff
@@ -344,7 +344,13 @@ func getReportPeriod(now time.Time, logger log.FieldLogger, report *metering.Rep
 // according the report's schedule. If the next scheduled reporting period
 // hasn't elapsed, runReport will requeue the resource for a time when
 // the period has elapsed.
+// Report expiration is checked before running, report is deleted if expired.
 func (op *Reporting) runReport(logger log.FieldLogger, report *metering.Report) error {
+	// check if the report is expired
+	if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
+		deleteReport(logger, report, op)
+		return nil
+	}
```

Comment:
I need to double-check how we currently process Report objects, but I would expect this kind of functionality to live higher up like in `syncReport` or `handleReport`.

Reply:
ok, let me know, I can lift it up as needed, if it's a bad place for it logically.

Reply:
Moved to `syncReport()` and now uses `op.deleteReport()`